### DISCUSSION TITLE
focus:shadow-outline isnt present in tailwind

### DISCUSSION THREAD

**Discussion by rockwellll** [tailwindlabs/tailwindcss#3668] [Help]
Hi, so i have a basic tailwind installation and configured. Everything seems to work fine.

Except when i try to do `focus:outline-none` it doesn't work. I also cannot see it in the generated css file. Any ideas why?

this basic code 

```slim
 = f.email_field :email, :class => "p-2 rounded-lg focus:outline-none mt-1 focus:shadow-outline"
```

in fact, it isnt working this way

```slim

f.email_field :email, :class => "p-2 rounded-lg focus:outline-none mt-1 shadow-outline"

```

**Comment by simonswiss** [comment]
Hey!

If you're using Tailwind CSS v2.0, the `focus:shadow-outline` is not a thing anymore. You can replace this with `focus:ring`, and use the power of the new ring utilities to create complex focus ring styles.

Here's the part of the [Upgrading to Tailwind CSS v2.0 guide](https://tailwindcss.com/docs/upgrading-to-v2#replace-shadow-outline-and-shadow-xs-with-ring-utilities) that is relevant.

Hope it helps 👍
